step-api/step-api-json-schema/src/main/java/step/handlers/javahandler/jsonschema/DefaultJsonSchemaFieldProcessor.java

```diff
@@ -45,9 +45,13 @@ public boolean applyCustomProcessing(Class<?> objectClass, Field field, FieldMet
         // 1. extract field parameters (name, required, default value etc)
         String type = JsonInputConverter.resolveJsonPropertyType(fieldMetadata.getType());
         String parameterName = fieldMetadata.getFieldName();
-        if (fieldMetadata.isRequired()) {
-            requiredPropertiesOutput.add(parameterName);
-        }
+
+        // moved to JsonSchemaCreator to be always applied (also for custom field processors)
+//        if (fieldMetadata.isRequired()) {
+//            requiredPropertiesOutput.add(parameterName);
+//        }
+
+        // TODO: default values should also be applied in all processors, but no in DefaultJsonSchemaFieldProcessor only
```

Comment:
@iegorov777 Do you plan to implement this todo as part of this PR i.e. should we wait to before merging it? Can the above comments be removed?

Reply:
@jeromecomte 
No, I plan to resolve all these todo's during the mass refactoring of serializers/deserializers in further release.